crates/consensus/src/transaction/typed.rs

```diff
@@ -0,0 +1,147 @@
+use alloy_network::Transaction;
+
+use crate::{TxEip1559, TxEip2930, TxEip4844, TxLegacy, TxType};
+
+/// The TypedTransaction enum represents all Ethereum transaction request types.
+///
+/// Its variants correspond to specific allowed transactions:
+/// 1. Legacy (pre-EIP2718) [`TxLegacy`]
+/// 2. EIP2930 (state access lists) [`TxEip2930`]
+/// 3. EIP1559 [`TxEip1559`]
+/// 4. EIP4844 [`TxEip4844`]
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum TypedTransaction {
```

Comment:
Should this impl RLP?

Reply:
unsure, do we want it? cc @mattsse perhaps